[MLIR] Update lhlo.const to linalg lowering to use affine.store inste…

bondhugula opened the pull request:
…ad of std.store

The xla_lhlo.const lowering uses std.store to store a constant to
0-d memrefs. Update it to affine.store since such an access is trivially
affine (no indices). An affine.store can always be lowered to std.store.

joker-eph commented:
@bondhugula the CLA-related message is in general an indication that the commit email isn't the usual one

bondhugula committed:
[MLIR] Update lhlo.const to linalg lowering to use affine.store instead of std.store

The xla_lhlo.const lowering uses std.store to store a constant to
0-d memrefs. Update it to affine.store since such an access is trivially
affine (no indices). An affine.store can always be lowered to std.store.

bondhugula commented:
@googlebot I fixed it.

bondhugula commented:
> @bondhugula the CLA-related message is in general an indication that the commit email isn't the usual one

Thanks, fixed.

bondhugula commented:
Ping @pifon2a - thanks.

gbaned commented:
@bondhugula Can you please resolve conflicts? Thanks!

bondhugula commented:
> @bondhugula Can you please resolve conflicts? Thanks!

Done, thanks!

gbaned commented:
@bondhugula Still, conflicts appearing, can you please resolve those? Thanks!

bondhugula commented:
> @bondhugula Still, conflicts appearing, can you please resolve those? Thanks!

Done - now. (Looks like those conflicts happened in the time my rebased commit was tested.)
